java/org/apache/catalina/realm/RealmBase.java

```diff
@@ -756,10 +756,11 @@ public void backgroundProcess() {
                 }
 
                 boolean matched = false;
-                for(int k=0; k < patterns.length && !matched; k++) {
+                for(int k = 0; k < patterns.length; k++) {
```

Comment:
Why not make it foreach directly?

Reply:
I wanted to make a separate PR for all possible foreach loops later

Reply:
see #273